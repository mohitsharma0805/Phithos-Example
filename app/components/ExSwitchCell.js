import React from 'react';
import { View, StyleSheet, Text, Switch } from 'react-native';
import config from '../config';

const ExSwitchCell = ({ switchValue, onSwitchValueChange }) => {
  const switchValueChanged = (value) => onSwitchValueChange?.(value);
  return (
    <View style={styles.container}>
      <View style={styles.wrapper}>
        <View style={styles.textWrapper}>
          <Text style={styles.title}>Banner Status</Text>
        </View>
        <Switch
          style={styles.switch}
          value={switchValue}
          onValueChange={switchValueChanged}
          trackColor={{
            true: config.colors.appTheme,
            false: null,
          }}
        />
      </View>
    </View>
  );
};

export default ExSwitchCell;

const styles = StyleSheet.create({
  container: { backgroundColor: config.colors.white },
  wrapper: { flexDirection: 'row', alignItems: 'center', height: 60 },
  textWrapper: { flex: 1 },
  title: {
    marginLeft: 16,
    paddingVertical: 2,
    fontSize: 17,
  },
  switch: { marginRight: 20, marginLeft: 16 },
});
