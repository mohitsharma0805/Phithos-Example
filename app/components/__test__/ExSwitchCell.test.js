import React from 'react';
import { shallow } from 'enzyme';
import { View, Text, Switch } from 'react-native';
import ExSwitchCell from '../ExSwitchCell';

describe('ExSwitchCell', () => {
  describe('Rendering', () => {
    const wrapper = shallow(<ExSwitchCell />);
    it('renders the container view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(View).length).toEqual(3);
      expect(wrapper.find(Text).length).toEqual(1);
      expect(wrapper.find(Switch).length).toEqual(1);
    });
  });
});