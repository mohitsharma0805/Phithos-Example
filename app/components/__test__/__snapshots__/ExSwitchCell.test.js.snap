// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExSwitchCell Rendering renders the container view 1`] = `ShallowWrapper {}`;
